rand_distr/src/pert.rs

```diff
@@ -99,7 +99,15 @@ where
 
         let range = max - min;
         let mu = (min + max + shape * mode) / (shape + F::from(2.).unwrap());
-        let v = if mu == mode {
+
+        // We need to check if `mu == mode`, but due to various rounding errors
+        // we can't use a direct comparison. The epsilon used for the
+        // comparison is twice the machine epsilon scaled by the largest input
+        // to the whole calculation.
+        //
+        // https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/
+        let epsilon = shape.max(min.abs()).max(max.abs()) * F::epsilon() * (F::one() + F::one());
```

Comment:
Must you call this complicated expression `epsilon`? And why does it involve `shape` anyway?